pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -250,35 +245,25 @@ func buildSidecarVirtualHostsForVirtualService(
 	return out
 }
 
-func buildSidecarVirtualHostsForService(
-	serviceRegistry map[host.Name]*model.Service,
-	hashByService map[host.Name]map[int]*networking.LoadBalancerSettings_ConsistentHashLB,
+func buildSidecarVirtualHostForService(svc *model.Service,
```

Comment:
@hzxuzhonghu This is the main change left over from previous PR i.e. `buildSidecarVirtualHostForService` for single service as its name suggests and move inside the loop above

Reply:
LGTM